tests/MongoDB.Bson.TestHelpers/JsonDrivenTests/JsonDrivenTestCaseFactory.cs

```diff
@@ -65,14 +52,44 @@ protected virtual IEnumerable<BsonDocument> ReadJsonDocuments()
         {
             return
                 Assembly.GetManifestResourceNames()
-                .Where(path => ShouldReadJsonDocument(path))
-                .Select(path => ReadJsonDocument(path));
+                    .Where(path => ShouldReadJsonDocument(path))
+                    .Select(path => ReadJsonDocument(path));
         }
 
         protected virtual bool ShouldReadJsonDocument(string path)
         {
-            return path.StartsWith(PathPrefix) && path.EndsWith(".json");
+            var prefixes = GetPathPrefixes();
```

Comment:
`GetPathPrefixes` is going to be called over and over again for EVERY path.

Is that a performance problem?

Reply:
I don't think so since it's for tests. And there is no huge logic inside this method as well as there is no a huge number of tests. From my perspective, any performance benefit which we can have here is not worth it